### PR TITLE
[MRG] Fix timezone bugs

### LINKED ISSUE #322: Timezone issues in DB
These tests fail, due to what seem to be timezone issues (the hour is 5 hours ahead/behind, I'm in UTC-5). This probably doesn't show up if the system clock is in UTC.

```
============================================================================================= test session starts ==============================================================================================
platform darwin -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/aapeli/git/couchers/app/backend/src, configfile: pytest.ini
plugins: pythonpath-0.7.3
collected 74 items
run-last-failure: no previously failed tests, not deselecting items.

src/tests/test_api.py ................                                                                                                                                                                   [ 21%]
src/tests/test_auth.py ......                                                                                                                                                                            [ 29%]
src/tests/test_bug_tool.py ...                                                                                                                                                                           [ 33%]
src/tests/test_conversations.py .......F....F............                                                                                                                                                [ 67%]
src/tests/test_db.py ....                                                                                                                                                                                [ 72%]
src/tests/test_email.py .......                                                                                                                                                                          [ 82%]
src/tests/test_media.py .                                                                                                                                                                                [ 83%]
src/tests/test_models.py ...                                                                                                                                                                             [ 87%]
src/tests/test_requests.py .........                                                                                                                                                                     [100%]

=================================================================================================== FAILURES ===================================================================================================
___________________________________________________________________________________________ test_get_group_chat_info ___________________________________________________________________________________________

db = sessionmaker(class_='Session', bind=Engine(postgresql://postgres:***@localhost:6544/postgres), autoflush=True, autocommit=False, expire_on_commit=True)

>   ???
E   assert datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) <= datetime.datetime(2020, 11, 21, 13, 35, 25, 179729)
E    +  where datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) = <bound method Timestamp.ToDatetime of seconds: 1605983724\nnanos: 998222000\n>()
E    +    where <bound method Timestamp.ToDatetime of seconds: 1605983724\nnanos: 998222000\n> = seconds: 1605983724\nnanos: 998222000\n.ToDatetime
E    +      where seconds: 1605983724\nnanos: 998222000\n = group_chat_id: 1\ntitle: "Test title"\nmember_user_ids: 2\nmember_user_ids: 1\nadmin_user_ids: 1\nonly_admins_invite: true\n...  author_user_id: 1\n  time {\n    seconds: 1605983725\n    nanos: 59876000\n  }\n  text {\n    text: "Test message 2"\n  }\n}\n.created
E    +  and   datetime.datetime(2020, 11, 21, 13, 35, 25, 179729) = <built-in method now of type object at 0x10c6da668>()
E    +    where <built-in method now of type object at 0x10c6da668> = datetime.now

/app/src/tests/test_conversations.py:395: AssertionError
______________________________________________________________________________________________ test_send_message _______________________________________________________________________________________________

db = sessionmaker(class_='Session', bind=Engine(postgresql://postgres:***@localhost:6544/postgres), autoflush=True, autocommit=False, expire_on_commit=True)

>   ???
E   assert datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) <= datetime.datetime(2020, 11, 21, 13, 35, 30, 214964)
E    +  where datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) = <bound method Timestamp.ToDatetime of seconds: 1605983730\nnanos: 196730000\n>()
E    +    where <bound method Timestamp.ToDatetime of seconds: 1605983730\nnanos: 196730000\n> = seconds: 1605983730\nnanos: 196730000\n.ToDatetime
E    +      where seconds: 1605983730\nnanos: 196730000\n = message_id: 2\nauthor_user_id: 1\ntime {\n  seconds: 1605983730\n  nanos: 196730000\n}\ntext {\n  text: "Test message 1"\n}\n.time
E    +  and   datetime.datetime(2020, 11, 21, 13, 35, 30, 214964) = <built-in method now of type object at 0x10c6da668>()
E    +    where <built-in method now of type object at 0x10c6da668> = datetime.now

/app/src/tests/test_conversations.py:584: AssertionError
=========================================================================================== short test summary info ============================================================================================
FAILED src/tests/test_conversations.py::test_get_group_chat_info - assert datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) <= datetime.datetime(2020, 11, 21, 13, 35, 25, 179729)
FAILED src/tests/test_conversations.py::test_send_message - assert datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) <= datetime.datetime(2020, 11, 21, 13, 35, 30, 214964)
=================================================================================== 2 failed, 72 passed in 66.76s (0:01:06) ====================================================================================
```

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #322 

Changes places that use raw datetime to use UTC-based utility functions in `couchers.utils`

### CODE REVIEW

File: app/backend/src/couchers/servicers/requests.py [hunk 1]
```diff
@@ -74,7 +73,7 @@ def CreateHostRequest(self, request, context):
             if request.from_date > request.to_date:
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.DATE_FROM_AFTER_TO)
 
-            if request.to_date < datetime.datetime.now().strftime("%Y-%m-%d"):
+            if request.to_date < now().strftime("%Y-%m-%d"):
```
Comment (amdwit) [inline app/backend/src/couchers/servicers/requests.py]: I would check whether the request dates are at or after the date in timezone UTC-12
taking into account the date at the location of the host does not add enough business value.

**Review by kalvdans — COMMENTED** [review]
Could we annotate, with comments in the code, which timezone is used in the API calls and the database? Then it would be easier to review and possibly tick off some TODOs in the code as well.

**Comment by aapeliv** [comment]
@kalvdans, I've added a comment in the style section of the contributor guide. Do you think there is another good place to comment on this?

**Comment by kalvdans** [comment]
> @kalvdans, I've added a comment in the style section of the contributor guide. Do you think there is another good place to comment on this?

I've pushed https://github.com/Couchers-org/couchers/pull/324/commits/8120e81f55f18ffa7bfc16ce3a2549d9b3ca09f7 to show how I would like it to be documented, i.e. inline in every place in the API description and database scheme description. (luckily Google.Protobuf.WellKnownTypes.Timestamp is timezoneless)

**Comment by kalvdans** [comment]
Could we add the constraint from https://github.com/BurntSushi/nfldb/issues/101#issue-106911118-permalink to our database tables? Then we would get an error if someone inserts non-utc data into the database. I don't know how to translate that constraint to sqlalchemy though.

File: docs/contributing.md [hunk 2]
```diff
@@ -68,4 +68,6 @@ Additionally, we strive to use the ["Google" docstring format](https://sphinxcon
 
 Please use double quotes `"`, instead of single quotes `'`.
 
-In the **frontend**, make sure to run `yarn format` in app/frontend before asking for a review.
+In the **frontend**, make sure to run `yarn format` in `app/frontend` before asking for a review.
+
+The server should operate in UTC, and we should always use datetimes with timezones. Please use the utility functions in `couchers.utils`. For now we pretend the whole world operates in UTC, which will of course have to change soon given we're going to care about local times. How this responsibility is divided up between the frontend and backend is not entirely clear yet though.
```
Comment (amdwit) [inline docs/contributing.md]: It does not matter what timezone the server uses as we should avoid using functionality that parses or formats using (server-)local timezone.

File: app/backend/src/couchers/servicers/requests.py (hunk 1 above)
Comment (kalvdans) [reply app/backend/src/couchers/servicers/requests.py]: I've fixed the comparison accordingly, please have another look.

File: docs/contributing.md (hunk 2 above)
Comment (kalvdans) [reply docs/contributing.md]: I've removed the text but kept the formatting of app/frontend. Resolving.

**Review by kalvdans — APPROVED** [review]
I'm approving, basically since I did a hostile takeover of the PR :D
Hope you agree to my changes.

**Comment by aapeliv** [comment]
Looks good. I'll merge later

**Pull request merged**